target/scripts/share-inbox.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# $1: The source account name
+# $2: The account name of who receives access
+# $3, $4 and so on: list of permissions - one of: lookup read write write-seen write-deleted insert post expunge
+# Call me like this: share_inbox.sh office bob lookup read
+
+DOMAIN=$(hostname -d)
```

Comment:
I will try review this properly later, but this caught my eye.

See the `targer/scripts/helpers/dns.sh` and how we source it to be used elsewhere (_via [`helpers/index.sh`](https://github.com/docker-mailserver/docker-mailserver/blob/master/target/scripts/helpers/index.sh)_).
- During startup scripts, so you can access the [`$DOMAINNAME`](https://github.com/docker-mailserver/docker-mailserver/blob/2b330fdc493d983f54a8f8868393d496e1e5a611/target/scripts/helpers/dns.sh#L64) variable (_which may not always be derived by `hostname -d`_).
- For `setup ...` commands which this seems to be, [source the helper index explicitly](https://github.com/docker-mailserver/docker-mailserver/blob/2b330fdc493d983f54a8f8868393d496e1e5a611/target/bin/open-dkim#L4).

The location of this script seems misplaced. `target/bin/` would be correct for user run commands, which can be hooked up with our `setup` CLI. Look at others for common structure, along with help message.

---

This would only support the domain name of the MX host which probably is not intentional either?

We keep track of all domains internally via `/etc/postfix/vhost` that is [created during startup (_at least for non-LDAP setups_)](https://github.com/docker-mailserver/docker-mailserver/blob/2b330fdc493d983f54a8f8868393d496e1e5a611/target/scripts/helpers/postfix.sh#L20-L84). You can see it being used by a `setup config dkim` command here (_which later loops through each domain collected to process_):

https://github.com/docker-mailserver/docker-mailserver/blob/2b330fdc493d983f54a8f8868393d496e1e5a611/target/bin/open-dkim#L101-L120

---

Final consideration is that the feature supporting each domain configured (_The MX host domain itself doesn't have to have any mail accounts related to it_), may not want to enable it for all domains available.

This would then need some sort of optional config support (_could be a separate PR_) to opt-in / opt-out domains for the feature. [Relayhost support has opt-out](https://github.com/docker-mailserver/docker-mailserver/blob/2b330fdc493d983f54a8f8868393d496e1e5a611/target/scripts/helpers/relay.sh) but is likely to be refactored into a simpler opt-in config instead.

The `setup` relay commands [provide some similar control](https://github.com/docker-mailserver/docker-mailserver/blob/master/target/bin/addrelayhost) and that may be appropriate. Note that these commands just manage user config files, and a separate startup helper script would make the `doveadm` call, which `check-for-changes.sh` can also support. Although I'm not sure how well this would work.

---

One other concern is how it's handled with removal of accounts, or unsharing a user. Most other features use a naive approach that works with `check-for-changes.sh` to use the user config files as input for recreating the entire desired state from scratch.

This would depend how `doveadm acl add` integrates with that (_other changes losing persistence of this feature_, or if this data persists differently_).